### PR TITLE
Fix for exception  thrown from connection.py if resp_body is None or empty from method do_http

### LINKED ISSUE #81: Exception is thrown from connection.py if resp_body is None or empty from method 'do_http'
Exception is thrown from connection.py if resp_body is None or empty from method 'do_http'  
Exception:
Traceback (most recent call last):
  File "examples/certificates.py", line 52, in <module>
    certificate.delete()
  File "/home/vverma/.pyenv/versions/3.7.6/lib/python3.7/site-packages/simplivity/resources/certificates.py", line 66, in delete
    self._client.do_delete(resource_uri, timeout)
  File "/home/vverma/.pyenv/versions/3.7.6/lib/python3.7/site-packages/simplivity/resources/resource.py", line 283, in do_delete
    task, body = self._connection.delete(uri, custom_headers=custom_headers)
  File "/home/vverma/.pyenv/versions/3.7.6/lib/python3.7/site-packages/simplivity/connection.py", line 172, in delete
    return self.__do_rest_call('DELETE', uri, {}, custom_headers=custom_headers)
  File "/home/vverma/.pyenv/versions/3.7.6/lib/python3.7/site-packages/simplivity/connection.py", line 203, in __do_rest_call
    custom_headers=custom_headers)
  File "/home/vverma/.pyenv/versions/3.7.6/lib/python3.7/site-packages/simplivity/connection.py", line 86, in do_http
    json_body = json.loads(resp_body.decode('utf-8'))
  File "/home/vverma/.pyenv/versions/3.7.6/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/home/vverma/.pyenv/versions/3.7.6/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/vverma/.pyenv/versions/3.7.6/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None

### PR DESCRIPTION
**Pull request by svt-vverma**
### Description
Fix for exception  thrown from connection.py if resp_body is None or empty from method do_http

### Issues Resolved
Fixes #81 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass(`$ tox`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: simplivity/connection.py [hunk 1]
```diff
@@ -77,18 +77,20 @@ def do_http(self, method, path, body, custom_headers=None, login=False):
...
 
+        json_body = None
         try:
             connection = self.get_connection()
             connection.request(method, full_path, body, http_headers)
             resp = connection.getresponse()
             resp_body = resp.read()
             connection.close()
-            json_body = json.loads(resp_body.decode('utf-8'))
+            if resp_body:
```
Comment (sijeesh) [inline simplivity/connection.py:87]: lets add a test case for null response

**Review by sijeesh — CHANGES_REQUESTED** [review]

File: simplivity/connection.py (hunk 1 above)
Comment (svt-vverma) [reply simplivity/connection.py:87]: test case added as per suggestion

**Review by sijeesh — APPROVED** [review]
LGTM
